File: backend/src/main/java/com/valeo/kanban/dto/response/ColumnDto.java
package com.valeo.kanban.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ColumnDto {
    private Long id;
    private String name;
    private double position;
    // Named "gated" so Lombok's isGated() getter and the field share one Jackson property
    @JsonProperty("isGated")
    private boolean gated;
    private Integer wipLimit;
    @Builder.Default
    private List<TaskDto> tasks = new ArrayList<>();
}

File: frontend/src/app/core/models/column.dto.ts
import { TaskDto } from './task.dto';

export interface ColumnDto {
  id: number;
  name: string;
  position: number;
  isGated: boolean;
  wipLimit?: number;
  tasks: TaskDto[];
}

export interface ColumnCreateRequest {
  name: string;
  position: number;
  isGated: boolean;
  wipLimit?: number;
}

export interface ColumnReorderRequest {
  columnId: number;
  newPosition: number;
}
